asynciterator.ts

```diff
@@ -309,6 +309,35 @@ export class AsyncIterator<T> extends EventEmitter {
     return '';
   }
 
+  /**
+    Consume all remaining items of the iterator into an array that will be returned asynchronously.
+    @param {object} [options] Settings for array creation
+    @param {integer} [options.limit] The maximum number of items to place in the array.
+   */
+  toArray(options?: { limit?: number }): Promise<T[]> {
+    return new Promise<T[]>((resolve, reject) => {
+      // Collect items in an array
+      const items: T[] = [];
+      let resolved = false;
+
+      this.on('error', reject);
+      this.on('data', item => {
+        if (!resolved)
+          items.push(item);
+
+        // Resolve earlier if a limit was set and we have reached that limit
+        if (options?.limit && items.length === options.limit) {
```

Comment:
```suggestion
        if (items.length > options?.limit) {
```
wouldn't want to dream someone setting a float limit, but yeah 😅

```javascript
[
    -5,    -1,        -0,
     0,   NaN, undefined,
  null, false,     "foo",
   0.1,     1,         3,
     4,     4.5,       6, // only this line should succeed
].map(limit => limit > 3)
// [
//   false, false, false,
//   false, false, false,
//   false, false, false,
//   false, false, false,
//   true,  true,  true,
// ]
```

Reply:
I did it like this because the TS compiler complains about comparing with `undefined`.
Should I suppress it with `@ts-ignore`, or keep the truthy `options?.limit` check?

Reply:
Removing `options?.limit` made it simpler to implement the zero-limit behaviour, so I went with the `@ts-ignore` for now. Can still change it if needed.

Reply:
Yeah; I think TypeScript is 100% right to warn here, because there are many edge cases. However, we have checked them all, and we have the right behavior.